server/pbench/bin/index-pbench

```diff
@@ -322,6 +330,168 @@ class ResultData(object):
         return datafiles
 
 
+    @staticmethod
+    def gen_result(ts):
+        """ ts is a list of tuples: singlets in the throughput case, doublets in the latency case.
+        """
+        for datum in ts:
+            yield datum
+        return
+
+    @staticmethod
+    def gen_sources(results):
+        # There should always be a 'parameters' block.
+        # There is no reason for this to be an array, but it is: we
+        # just pick the 0^th (and only element) as the metadata for
+        # the document.
+        md = results['parameters']['benchmark'][0]
```

Comment:
Can we wrap this in a `try` block to catch badly formed results?

Reply:
I'll take a stab at it.